Marlin/Marlin_main.cpp

```diff
@@ -4478,6 +4481,133 @@ inline void gcode_G92() {
+ *       
+ * NOTE - A minimum PWM frequency of 50 Hz is needed.  All of the 
+ *        pre-scaler factors for timers 2, 3, 4 & 5 are acceptable. 
+ *
+ *  SPINDLE_ENABLE_PIN needs an external pullup or else could turn on 
+ *  spindle  during power up or when connecting to the host (usually 
+ *  goes through a reset which sets all I/O pins to tristate)
+ *
+ *  PWM duty cycle goes from 0 (off) to 255 (always on) 
+ * 
+ */  
+#if ENABLED(SPINDLE_ENABLE)
+
+  inline void delay_for_spindle_power_up() {
+    refresh_cmd_timeout();                                                       // wait for spindle to come up to speed
+    if (SPINDLE_POWER_UP_DELAY > 0)  while (PENDING(millis(), SPINDLE_POWER_UP_DELAY * 1000 + previous_cmd_ms)) idle();
+    else                             while (PENDING(millis(), 1000 + previous_cmd_ms)) idle();         // allow at least 1 second for power up
+  }
+  
+  inline void delay_for_spindle_power_down() {
+    refresh_cmd_timeout();                                                       // wait for spindle to come up to speed
+    if (SPINDLE_POWER_DOWN_DELAY >= 0)  while (PENDING(millis(), SPINDLE_POWER_DOWN_DELAY * 1000 + previous_cmd_ms + 1)) idle(); 
+    else                                while (PENDING(millis(), 1000 + previous_cmd_ms)) idle();    // allow at least 1 second for power down
+  }
+
+  inline void gcode_M3_M4() {
+
+    // this is the "pretty" version.  It accepts RPM as input
+    #if ENABLED(SPINDLE_SPEED)
+  
+      // use gcode_M4() to get the points needed to compute the RPM vs OCRxx_VAL line
+  
+      // Our final value for OCRxx_VAL is an unsigned 8 bit value between 0 and 255 which usually means uint8_t.
+      // Went to uint16_t because some of the uint8_t calculations would sometimes give 1000 0000 rather than 1111 1111.
+      // Then needed to AND the uint16_t result with 0xffff to make sure we only wrote the byte of interest.
+  
+      float spindle_speed = code_seen('S') ? code_value_float() : 0; 
+      if (spindle_speed == 0) { 
+        stepper.synchronize();   // wait until previous movement commands (G0, G1, G2 & G3) have completed before turning spindle off
+        WRITE(SPINDLE_ENABLE_PIN, !SPINDLE_ENABLE_INVERT);  //turn spindle off (active low);
+        delay_for_spindle_power_down();
+      }
+      else {
+        int16_t OCRxx_VAL = 2.55 * (spindle_speed - RPM_INTERCEPT)/RPM_SLOPE;      // convert RPM to PWM duty cycle
+        float temp = 2.55 * (spindle_speed - RPM_INTERCEPT)/RPM_SLOPE;
+        if (temp >= 255)                OCRxx_VAL = 255;                                        //limit to max the Atmel PWM will support
+        if (spindle_speed <= RPM_MIN)   OCRxx_VAL = 2.55 * (RPM_MIN - RPM_INTERCEPT)/RPM_SLOPE; // minimum setting
+        if (spindle_speed >= RPM_MAX)   OCRxx_VAL = 2.55 * (RPM_MAX - RPM_INTERCEPT)/RPM_SLOPE; // limit to max RPM
+        if (SPINDLE_SPEED_INVERT==true) OCRxx_VAL = 255 - OCRxx_VAL ; 
```

Comment:
You can leave off `==true` and there should be no extra space before the semicolon.